components/raftstore/src/store/fsm/apply.rs

```diff
@@ -5491,6 +5525,88 @@ mod tests {
         });
     }
 
+    #[test]
+    fn test_exec_observer() {
```

Comment:
please write some comments about what's the test points of this test and explain which parts of the new code are tested.